File: backend/WedaLK/demo/src/main/java/lk/wedalk/auth/service/AuthService.java
package lk.wedalk.auth.service;

/**
 * AuthService.java — Authentication Business Logic
 *
 * This file should contain:
 * - @Service annotation
 * - Inject UserRepository, PasswordEncoder, JwtTokenProvider (future)
 * - Methods:
 * - AuthResponse register(RegisterRequest request)
 * - Check if email already exists
 * - Hash the password
 * - Create User entity and save
 * - Generate JWT token
 * - Return AuthResponse
 * - AuthResponse login(LoginRequest request)
 * - Find user by email
 * - Verify password
 * - Generate JWT token
 * - Return AuthResponse
 *
 * Purpose:
 * Handles user registration and authentication logic.
 */

File: frontend/src/services/authService.js
/**
 * authService.js — Authentication Service
 *
 * This file should contain:
 * - Import apiClient from './apiClient'
 * - Functions:
 *     - login(email, password) → POST /auth/login → returns AuthResponse
 *     - register(registerData) → POST /auth/register → returns AuthResponse
 *     - logout() → clear token from localStorage, redirect to /login
 *     - getCurrentUser() → decode JWT or call user endpoint
 *     - isAuthenticated() → check if valid token exists
 *
 * Purpose:
 *   Centralizes all authentication-related API calls.
 *   Used by LoginPage, RegisterPage, AuthContext.
 *
 * Export: { login, register, logout, getCurrentUser, isAuthenticated }
 */
